src/server.c

```diff
@@ -193,11 +196,512 @@ struct redisServer server; /* Server global state */
+    {"get",configGetCommand,3,
+     "admin ok-loading ok-stale no-script"},
+
+    {"resetstat",configResetStatCommand,2,
+     "admin no-script"},
+
+    {"rewrite",configRewriteCommand,2,
+     "admin no-script"},
+
+    {"help",configHelpCommand,2,
+     ""},
+
+    {NULL},
+};
+
+struct redisCommand xinfoSubcommands[] = {
+    {"consumers",xinfoCommand,4,
+     "read-only random @stream",
+     {{"read",
+       KSPEC_BS_INDEX,.bs.index={2},
+       KSPEC_FK_RANGE,.fk.range={0,1,0}}}},
+
+    {"groups",xinfoCommand,3,
+     "read-only @stream",
+     {{"read",
+       KSPEC_BS_INDEX,.bs.index={2},
+       KSPEC_FK_RANGE,.fk.range={0,1,0}}}},
+
+    {"stream",xinfoCommand,-3,
+     "read-only @stream",
+     {{"read",
+       KSPEC_BS_INDEX,.bs.index={2},
+       KSPEC_FK_RANGE,.fk.range={0,1,0}}}},
+
+    {"help",xinfoCommand,2,
+     "@stream"},
+
+    {NULL},
+};
+
+struct redisCommand xgroupSubcommands[] = {
+    {"create",xgroupCommand,-5,
+     "write use-memory @stream",
+     {{"write",
+       KSPEC_BS_INDEX,.bs.index={2},
+       KSPEC_FK_RANGE,.fk.range={0,1,0}}}},
+
+    {"setid",xgroupCommand,5,
+     "write @stream",
+     {{"write",
+       KSPEC_BS_INDEX,.bs.index={2},
+       KSPEC_FK_RANGE,.fk.range={0,1,0}}}},
+
+    {"destroy",xgroupCommand,4,
+     "write @stream",
+     {{"write",
+       KSPEC_BS_INDEX,.bs.index={2},
+       KSPEC_FK_RANGE,.fk.range={0,1,0}}}},
+
+    {"createconsumer",xgroupCommand,5,
+     "write use-memory @stream",
+     {{"write",
+       KSPEC_BS_INDEX,.bs.index={2},
+       KSPEC_FK_RANGE,.fk.range={0,1,0}}}},
+
+    {"delconsumer",xgroupCommand,5,
+     "write @stream",
+     {{"write",
+       KSPEC_BS_INDEX,.bs.index={2},
+       KSPEC_FK_RANGE,.fk.range={0,1,0}}}},
+
+    {"help",xgroupCommand,2,
+     "@stream"},
+
+    {NULL},
+};
+
+struct redisCommand commandSubcommands[] = {
+    {"count",commandCommand,2,
+     "ok-loading ok-stale @connection"},
+
+    {"info",commandCommand,-3,
+     "ok-loading ok-stale @connection"},
+
+    {"getkeys",commandCommand,-4,
+     "ok-loading ok-stale @connection"},
+
+    {"help",commandCommand,2,
+     ""},
+
+    {NULL},
+};
+
+struct redisCommand commandsSubcommands[] = {
+    {"count",commandsCommand,2,
+     "ok-loading ok-stale @connection"},
+
+    {"list",commandsCommand,2,
+     "ok-loading ok-stale @connection"},
+
+    {"info",commandsCommand,-2,
+     "ok-loading ok-stale @connection"},
+
+    {"getkeys",commandCommand,-4,
+     "ok-loading ok-stale @connection"},
+
+    {"help",commandsCommand,2,
+     ""},
+
+    {NULL},
+};
+
+struct redisCommand memorySubcommands[] = {
+    {"doctor",memoryCommand,2,
+     "random"},
+
+    {"stats",memoryCommand,2,
+     "random"},
+
+    {"malloc-stats",memoryCommand,2,
+     "random"},
+
+    {"purge",memoryCommand,2,
+     "random"},
+
+    {"usage",memoryCommand,-3,
+     "read-only",
+     {{"read",
+       KSPEC_BS_KEYWORD,.bs.index={2},
+       KSPEC_FK_RANGE,.fk.range={0,1,0}}}},
```

Comment:
it does - if there are only "range" specs it glues them together to create the legacy spec

Reply:
good thing you asked, i had a bug (used KSPEC_BS_KEYWORD instead of KSPEC_BS_INDEX)

Reply:
ok, so we can theoretically drop the legacy spec from the command table now, and rely on that instead?
maybe we should at least add a test for MEMORY USAGE (maybe using ACL)

Reply:
just to make sure you didn't miss it. please add a test for it using ACL

Reply:
i already added a test that uses COMMAND(S) GETKEY memory usage

that should be enough, no?

Reply:
I suppose.. I don't remember what I had in mind. 
Maybe I wanted to check that redis internals are using the legacy key spec properly rather than the COMMAND command Metadata. 
But actually GETKEYS does that too (uses the same internal mechanism) . 
So I suppose that's ok